pkg/k8s/pipelineRunFetcher.go

```diff
@@ -0,0 +1,87 @@
+package k8s
+
+import (
+	"fmt"
+
+	api "github.com/SAP/stewardci-core/pkg/apis/steward/v1alpha1"
+	stewardLister "github.com/SAP/stewardci-core/pkg/client/listers/steward/v1alpha1"
+	"github.com/pkg/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+// PipelineRunFetcher has methods to fetch PipelineRun objects from Kubernetes
+type PipelineRunFetcher interface {
+	pipelineRunNameFetcher
+	ByKey(key string) (*api.PipelineRun, error)
+}
+
+type pipelineRunNameFetcher interface {
+	ByName(namespace, name string) (*api.PipelineRun, error)
+}
```

Comment:
Better make two public single-method interfaces and an umbrella interface just combining the two.